### DISCUSSION TITLE
Which commands are working for DEEBOT T9?

### DISCUSSION THREAD

**Discussion by unclej84** [mrbungle64/node-red-contrib-ecovacs-deebot#7] [General]
Following commands have been successfully tested on DEEBOT T9(+):
- Charge
- Connect
- CustomArea
- Disconnect
- FindMe
- GetAutoEmpty
- GetBatteryState
- GetCleanSpeed
- GetCleanSum
- GetChargeState
- GetLifeSpan (combo element for filter, main brush and side brush)
- GetNetInfo
- GetPosition
- GetVolume
- MoveBackward
- MoveForward
- Pause
- PlaySound
- ResetLifeSpan
- Resume
- Spot
- SpotArea
- Stop

Not working:
- Edge
- MoveLeft
- MoveRight

Not tested yet:
- SetWaterLevel

In evaluation (command gets reply from robot with more output than requested):
- GetChargerPos (event "ChargePosition" seen unrelated to "GetChargerPos")
- GetCleanLogs ("CleanLog" returns empty array[0] and "LastCleanLogs" an object with "null" values)
- GetCleanState (returns "CleanReport" and twice "ChargeState", Issue #9)
- GetLifeSpan (see Issue #2 and #3)
- GetWaterLevel (returns "WaterBoxInfo" and "WaterLevel", Issue #8)
- GetWaterBoxInfo (returns "WaterBoxInfo" and "WaterLevel", Issue #8)

List will be updated.

History:
2021-05-18:
- CustomArea moved from "not tested" to "working
- Edge moved from "not tested" to "not working
- Pause moved from "not tested" to "working
- ResetLifeSpan moved from "not tested" to "working
- Resume moved from "not tested" to "working
- Spot moved from "not tested" to "working
- SpotArea moved from "not tested" to "working

2021-05-01:
- GetAutoEmpty added to "working"
- GetChargerPos added to "in evaluation"
- GetCleanLogs moved from "not working" to "in evaluation"
- GetNetInfo moved from "not working" to "working"
- GetVolume added to "working"

**Comment by mrbungle64** [comment]
The following commands should be removed as they have not worked on any model so far:

* GetNetInfo
* MoveTurnAround

The list might also be updated.

**Reply by unclej84** [reply]
List has been updated, PR #11 created

**Comment by mrbungle64** [comment]
@unclej84 

Some of the events are not fully implemented yet.
I will add them to ecovacs-deebot.js now.

**Reply by mrbungle64** [reply]
@unclej84 

I just added the follwing commands:
* Relocate
* GetAutoEmpty
* GetNetInfo
* GetPosition
* GetVolume

I also added event listeners for
* AutoEmpty
* ChargePosition
* CleanLog
* LastCleanLogs
* CleanSum
* DeebotPosition
* NetworkInfo
* Volume

**Comment by mrbungle64** [comment]
@unclej84 

I just realized that I mixed up `GetNetInfo` and `GetDeviceInfo` 😉 
I'll add `GetNetInfo` again

**Reply by unclej84** [reply]
Okay, I will add it back to the list - is this one of the events not yet fully implemented in ecovacs-deebot.js or is it simply not working for the T9?

**Reply by mrbungle64** [reply]
In this case I mean the Node-RED node.
I'll commit the changes in some minutes.
